src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -608,7 +608,28 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
 
                     if ((object)associatedPropertyOrEvent == null)
                     {
-                        diagnostics.Add(ErrorCode.ERR_OverrideNotExpected, overridingMemberLocation, overridingMember);
+                        bool suppressError = false;
+                        if (overridingMemberIsMethod || overridingMember.IsIndexer())
+                        {
+                            var parameterTypes = overridingMemberIsMethod
+                                ? ((MethodSymbol) overridingMember).ParameterTypes
+                                : ((PropertySymbol) overridingMember).ParameterTypes;
+
+                            foreach (var parameterType in parameterTypes)
+                            {
+                                if (parameterType.TypeSymbol.VisitType((typeSymbol, unused1, unused2) => typeSymbol.IsErrorType(), (object)null) != null)
```

Comment:
This is the cause of the build error. A change was merged yesterday that prohibits some uses of `==` on `TypeSymbol`.
The solution here is to add a cast on the left-hand-side operand: if ((object)... != null)`

```
2018-12-20T21:40:19.8558529Z F:\vsagent\6\s\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs(620,37): error CS0619: 'TypeSymbol.operator !=(TypeSymbol, TypeSymbol)' is obsolete: 'Use 'TypeSymbol.Equals(TypeSymbol, TypeSymbol, TypeCompareKind)' method.' [F:\vsagent\6\s\src\Compilers\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.csproj]
2018-12-20T21:40:19.8576187Z F:\vsagent\6\s\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs(749,37): error CS0619: 'TypeSymbol.operator ==(TypeSymbol, TypeSymbol)' is obsolete: 'Use 'TypeSymbol.Equals(TypeSymbol, TypeSymbol, TypeCompareKind)' method.' [F:\vsagent\6\s\src\Compilers\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.csproj]
2018-12-20T21:40:19.8576490Z F:\vsagent\6\s\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs(797,37): error CS0619: 'TypeSymbol.operator ==(TypeSymbol, TypeSymbol)' is obsolete: 'Use 'TypeSymbol.Equals(TypeSymbol, TypeSymbol, TypeCompareKind)' method.' [F:\vsagent\6\s\src\Compilers\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.csproj]
2018-12-20T21:40:19.8576729Z F:\vsagent\6\s\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs(831,37): error CS0619: 'TypeSymbol.operator ==(TypeSymbol, TypeSymbol)' is obsolete: 'Use 'TypeSymbol.Equals(TypeSymbol, TypeSymbol, TypeCompareKind)' method.' [F:\vsagent\6\s\src\Compilers\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.csproj]
```

Reply:
Or presumably I could use `! ... is null`?

Reply:
That would work too (but requires extra parens)